### PR TITLE
Feature/selenium3.0

### PR DESCRIPTION
**Pull request by ryantrevisol**
I was able to get the plugin working with Selenium 3.0.1, and it works well. However, on my local environment, I wasn't able to get the unit tests to work. In `SeleniumTest`, no RCs check in. In the InjectedTests, I'm getting
`java.lang.NoSuchMethodError: org.jvnet.hudson.test.HudsonTestCase$WebClient.closeAllWindows()V`

I'm assuming something is wrong with my environment, because when I skip tests in maven, it builds fine and I am using it on my jenkins instance to run tests.

### CODE REVIEW

**Comment by Mobrockers** [comment]
I am currently on vacation, so I won't be able to merge right now. I'll take a look probably some time next week. I don't think the injected tests should be falling because of your changes. Try upgrading the test harness perhaps? Try to get the unit tests working again as well, if you can. I know they're barely useful but it's all we have currently, and they should work too. Or if you could write better ones that work, even better.

**Comment by ryantrevisol** [comment]
I will take a look when I have time. I have a feeling the issues stem from me raising the Java level to 8 (a Selenium 3.0 requirement).

File: pom.xml [hunk 1]
```diff
@@ -16,12 +16,13 @@
...
-		<selenium.version>${selenium.short.version}.0</selenium.version>
-        <htmlunit.version>2.20</htmlunit.version>
+		<selenium.short.version>3.0</selenium.short.version>
+		<selenium.version>${selenium.short.version}.1</selenium.version>
+        <htmlunit.version>2.21</htmlunit.version>
+		<java.level>8</java.level>
 	</properties>
 	<artifactId>selenium</artifactId>
-	<version>2.53.2-SNAPSHOT</version>
+	<version>3.0.0-SNAPSHOT</version>
```
Comment (Mobrockers) [inline pom.xml]: By the way, you seem to have updated the plugin version manually in the pom file. You should not do this. Jenkins has a maven build-release plugin workflow where upon release the version number for the next release, and next snapshot version are provided. Please revert that part of your pull request.

**Comment by foxhound91** [comment]
Hi guys, I'm really interested on getting this 3.0 selenium version working...
I've been checking the Jenkins build log and the error says:
> Caused by: org.codehaus.plexus.compiler.CompilerException: invalid target release: 1.8

And looking the Jenkins settings at Environment variables says:
> JAVA_HOME	/opt/jdk/jdk1.7.latest

So the java version at Jenkins should be updated in order to get this build properly.
Do you agree with me?

**Comment by Mobrockers** [comment]
Yes, jenkins cannot currently build plugins for Java 8 without requesting the build environment be manually changed iirc. This does not however matter if you're building local, which also does not work according to ryan, as the test harness fails.

**Comment by ryantrevisol** [comment]
@Mobrockers so are we stuck until Jenkins allows Java 8 to build plugins? 

In the meantime, anyone who needs the plugin with Selenium 3.0.1 can download the one I built for myself here: 
https://drive.google.com/open?id=0B4tjUatFMp6-SlFHRjI0MjV0WE0

**Comment by Mobrockers** [comment]
I can request java 8 builds to be enabled, but until the tests are working it does not matter. I won't release a new version without working tests.

**Comment by janhollevoet** [comment]
I tried the build provided by ryantrevisol, but get this

jan 13, 2017 1:53:20 PM org.openqa.grid.internal.utils.SelfRegisteringRemote <init>
WARNING: error getting the parameters from the hub. The node may end up with wrong timeouts.Error with the JSON of the config : String hudson.plugins.selenium.JenkinsCapabilityMatcher could not be coerced to class: java.lang.Class

The node is not registered: INFO: No registration sent ( register = false )

**Comment by Mobrockers** [comment]
@ryantrevisol 3.0.1 has been merged in now, with working tests. I like the other changes you made such as the configurationbuilder and getUrl and would be interested in merging those too if you can fix up your pull request! If not, I'll see if I can fix the merge myself when I have time.

**Comment by ryantrevisol** [comment]
@Mobrockers Thanks for merging my stuff. Sorry I didn't see this. I am in the FontAwesome 5 repo and have been getting spammed with notifications from that.

**Pull request merged**